beacon-chain/db/filesystem/blob.go

```diff
@@ -295,7 +290,7 @@ func (bs *BlobStorage) processFolder(folder os.FileInfo, currentSlot, retentionS
 		return 0, err
 	}
 	var num int
-	if slot < (currentSlot - retentionSlots) {
+	if slot < (currentSlot - bs.retentionSlots) {
```

Comment:
If you follow `currentSlot` through to where it's computed in `tryPrune` you will see it's currently misnamed - it represents the oldest slot that should *not* be pruned. You need to either pass in the current slot in to Prune, or fix the name to reflect that it represents the oldest slot to not delete, and fix this comparison. My preference would be for the latter but it's a bit more work.